### PR TITLE
Allow calling Bugsnag.notify() with no arguments

### LINKED ISSUE #196: Calling Bugsnag.notify() with an undefined/null argument silently fails
If Bugsnag.notify() is called with no argument, or an undefined/null variable it will silently fail. The error is send to Bugsnag but dropped as it doesn't have an error class.

### PR DESCRIPTION
**Pull request by wordofchristian**
fixes #196

### CODE REVIEW

**Comment by snmaynard** [comment]
I think we should actually notify in this case

**Comment by wordofchristian** [comment]
Should we just send `"Bugsnag.notify() called with no arguments"` as the name?

**Comment by snmaynard** [comment]
as the message maybe. Can we put something like BugsnagError as the class?

File: src/bugsnag.js [hunk 1]
```diff
@@ -115,6 +118,12 @@
   // Notify Bugsnag about an error by passing in a `name` and `message`,
   // without requiring an exception.
   self.notify = function (name, message, metaData, severity) {
+    if (!name) {
+      name = "UnspecifiedBugsnagError";
```
Comment (eanakashima) [inline src/bugsnag.js]: Might be worth pulling this to the top of the file where we declare some other constant-esque variables?

**Comment by eanakashima** [comment]
👍  Looks good! One thought: since this is a manual call to Bugsnag.notify, maybe the error class doesn't need to have "Error" in it? A class name like `BugsnagNotify` might get closer to suggesting the typical severity.

**Pull request merged**